### PR TITLE
[MAINT] add workflow to check for f strings

### LINKED ISSUE #2528: Automate the Formatting stuff
Prompted by https://github.com/nilearn/nilearn/pull/2477#issuecomment-648905550

We have occasionally discussed this, but I think now is the time.

Nilearn contributors, managers, core developers juggle a lot of other responsibilities. We should automate the stuff that we can.

There are automated formatters & linters as well as cloud based code quality platforms that would automate a lot of minutia, making it a non-issue for contributors & reviewers. We can set up our custom for these tools & then we should start incorporating those.

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #2528 
Relates to https://github.com/orgs/nilearn/projects/2/views/1?pane=issue&itemId=30677086

Changes proposed in this pull request:

- add workflow to run flynt on main and PRs
- add flynt config to pyproject.toml
- run flynt

### CODE REVIEW

**Comment by Remi-Gau** [comment]
oh and it is already one the pre-commit hook that is run, this would just provide an official check in CI

**Review by ymzayek — APPROVED** [review]

File: nilearn/_utils/data_gen.py [hunk 1]
```diff
@@ -677,7 +677,7 @@ def generate_group_sparse_gaussian_graphs(n_subjects=5,
     topology = topology > 0
     assert (np.all(topology == topology.T))
     logger.log(
-        "Sparsity: {0:f}".format(1. * topology.sum() / (topology.shape[0]**2)),
+        f"Sparsity: {1.0 * topology.sum() / topology.shape[0] ** 2:f}",
```
Comment (bthirion) [inline nilearn/_utils/data_gen.py:680]:
```suggestion
        f"Sparsity: {1.0 * topology.sum() / topology.shape[0] ** 2 :f}",
``` would be more readable IMHO

**Comment by tsalo** [comment]
@Remi-Gau have you looked at `flake8-use-fstring`? I don't know if it's better or worse than `flynt`, but you'd be able to include it in the `flake8` workflow that already exists.

**Comment by Remi-Gau** [comment]
Was not aware of this one. Will investigate.

**Comment by Remi-Gau** [comment]
> @Remi-Gau have you looked at `flake8-use-fstring`? I don't know if it's better or worse than `flynt`, but you'd be able to include it in the `flake8` workflow that already exists.

OK as far I can tell the 2 are not completely mutually exclusive. Flynt will actually try to convert % and Format to f strings but can only do the most simple ones because it tries to be conservative.

The flake8 extension will flag % and format and potentially report them as error.

So the 2 could coexist but one will throw more errors than the other when used for Linting purposes.

I would suggest to start with Flynt and and then tackle the remaining non f strings. Then we can switch to flake8 extensions to ensure we stick to f strings from now on.

**Comment by tsalo** [comment]
👍 sounds good to me. Thanks for taking a look at the flake8 extension.

File: nilearn/_utils/data_gen.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/_utils/data_gen.py:680]: let's see what black has to say about this

**Comment by bthirion** [comment]
The PR LGTM, but I think lots of red CI.

**Comment by Remi-Gau** [comment]
> The PR LGTM, but I think lots of red CI.

yeah I will rebase to clean some of that

**Review by bthirion — COMMENTED** [review]
Now it mixed several things but LGTM overall.

**Comment by Remi-Gau** [comment]
yeah some files should not be in that diff: will fix

**Pull request merged**